llama_index/readers/weaviate/client.py

```diff
@@ -174,51 +185,29 @@ def _to_node(entry: Dict) -> Node:
 
 def _node_to_dict(node: Node) -> dict:
     node_dict = node.to_dict()
-    node_dict.pop("embedding")  # NOTE: stored outside of dict
-
-    # json-serialize the extra_info
-    extra_info = node_dict.pop("extra_info")
-    extra_info_str = ""
-    if extra_info is not None:
-        extra_info_str = json.dumps(extra_info)
-    node_dict["extra_info"] = extra_info_str
-
-    # json-serialize the node_info
-    node_info = node_dict.pop("node_info")
-    node_info_str = ""
-    if node_info is not None:
-        node_info_str = json.dumps(node_info)
-    node_dict["node_info"] = node_info_str
-
-    # json-serialize the relationships
-    relationships = node_dict.pop("relationships")
-    relationships_str = ""
-    if relationships is not None:
-        relationships_str = json.dumps(relationships)
-    node_dict["relationships"] = relationships_str
-
-    ref_doc_id = node.ref_doc_id
-    if ref_doc_id is not None:
-        node_dict["ref_doc_id"] = ref_doc_id
     return node_dict
 
 
 def _add_node(
     client: Any, node: Node, class_prefix: str, batch: Optional[Any] = None
 ) -> str:
     """Add node."""
-    node_dict = _node_to_dict(node)
-    vector = node.embedding
+    metadata = {}
+    metadata['text'] = node.get_text()
+    metadata['node_id'] = node.get_doc_id()
```

Comment:
could we error if either of these two keys are in extra info already?

Reply:
text and node_id shouldn't be in extra_info

Reply:
Yes since it's user specified